addon/ng2/tasks/uninstall.ts

```diff
@@ -0,0 +1,84 @@
+import * as Task from 'ember-cli/lib/models/task';
+import * as npmTask from 'ember-cli/lib/tasks/npm-task';
+import * as chalk from 'chalk';
+import * as path from 'path';
+import * as search from 'typings-core/dist/search';
+import * as typings from 'typings-core/dist/uninstall';
+import * as systemJS from '../utilities/systemjs-helper';
+
+module.exports = Task.extend({
+  run: function(options) {
+    this.packages = options.packages;
+
+    return this.uninstallProcedure();
+  },
+
+  uninstallProcedure: function() {
+    const that = this;
+
+    const NpmTask = new npmTask({
+      command: 'uninstall',
+      ui: this.ui,
+      analytics: this.analytics,
+      project: this.project,
+      startProgressMessage: 'Uninstalling packages: ' + this.packages,
+      completionMessage: 'Packages successfully uninstalled.'
+    });
+
+    return NpmTask.run({
+      packages: this.packages,
+      verbose: false
+    })
+    .then(() => this.removeFromSystemJSConfig(this.packages))
+    .then(() => this.uninstallTypings());
+  },
+
+  uninstallTypings: function() {
+    this.ui.startProgress(chalk.green('Uninstalling typings'), chalk.green('.'));
+    this.packages = this.packages.filter(p => !/node-sass|stylus|less|compass-importer/.test(p));
+    let typingsList = [];
+
+    return new Promise(resolve => {
```

Comment:
You don't need that, just `return Promise.all(...).then(...)`. `then()` returns a Promise.